spec/rspec/core/formatters_spec.rb

```diff
@@ -130,28 +130,47 @@ module RSpec::Core::Formatters
       end
     end
 
-    describe "#setup_default", "with profiling enabled" do
+    describe "#setup_default" do
       let(:setup_default) { loader.setup_default output, output }
 
-      before do
-        allow(RSpec.configuration).to receive(:profile_examples?) { true }
+      context "with a formatter that implements #message" do
+        it 'doesnt add a fallback formatter' do
+          allow(reporter).to receive(:registered_listeners).with(:message) { [:json] }
+          setup_default
+          expect(loader.formatters).to exclude(an_instance_of ::RSpec::Core::Formatters::MessageFormatter)
+        end
       end
 
-      context "without an existing profile formatter" do
-        it "will add the profile formatter" do
-          allow(reporter).to receive(:registered_listeners).with(:dump_profile) { [] }
+      context "without a formatter that implements #message" do
+        it 'adds a fallback for message output' do
+          allow(reporter).to receive(:registered_listeners).with(:message) { [] }
           setup_default
-          expect(loader.formatters.last).to be_a ::RSpec::Core::Formatters::ProfileFormatter
+          expect(loader.formatters.last).to be_a ::RSpec::Core::Formatters::MessageFormatter
```

Comment:
It seems brittle to specify that the message formatter is last.  Don't we just care that the formatters list includes it?